src/test/java/com/synopsys/integration/jenkins/JenkinsProxyHelperTest.java

```diff
@@ -0,0 +1,95 @@
+package com.synopsys.integration.jenkins;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.junit.jupiter.api.Test;
+
+import com.synopsys.integration.rest.proxy.ProxyInfo;
+
+public class JenkinsProxyHelperTest {
+
+    ProxyInfo proxyInfo;
+    List<Pattern> patterns = new ArrayList<>();
+
+    final String url_1 = "www.include.com";
+    final String url_2 = "www.exclude.com";
+
+    Pattern pattern_1 = Pattern.compile(url_1);
+    Pattern pattern_2 = Pattern.compile(url_2);
+
+    final String protocol = "https://";
+    final String proxyHost = "proxyHost";
+    final String proxyUsername = "proxyUsername";
+    final String proxyPassword = "proxyPassword";
+    final String ntlmDomain = "ntlmDomain";
+    final String ntlmWorkstation = "ntlmWorkstation";
+
+    @Test
+    public void testGetProxyInfoFromJenkins_NullJenkins() {
+        proxyInfo = JenkinsProxyHelper.getProxyInfoFromJenkins(protocol + url_1);
+        assertFalse(proxyInfo.getHost().isPresent());
+        assertEquals(0, proxyInfo.getPort());
+    }
+
+    @Test
+    public void testGetProxyInfo_EmptyUrlMatchingExclude() {
+        // The returning proxyInfo object should be equal to ProxyInfo.NO_PROXY_INFO
+        patterns.clear();
```

Comment:
Consider writing a @ Before method to clear the patterns arraylist or even to instantiate it-- relying on the developer to clear the patterns list every time is asking for them to forget to.

Alternatively, this class may be a good case for writing an @ ParameterizedTest see: https://github.com/blackducksoftware/polaris-common/blob/master/src/test/java/com/synopsys/integration/polaris/common/service/PolarisServiceTest.java#L84
For an example of what that might look like.